### PR TITLE
Added custom (own) translator

### PR DESCRIPTION
**Pull request by kubomikita**

### CODE REVIEW

**Comment by kubomikita** [comment]
is it ok ? can you merge this pull request ? thanks

File: src/DI/TranslationExtension.php [hunk 1]
```diff
@@ -53,6 +53,7 @@ public function getConfigSchema(): Nette\Schema\Schema
 				'dir' => Expect::string($builder->parameters['tempDir'] . '/cache/translation'),
 				'factory' => Expect::string(Symfony\Component\Config\ConfigCacheFactory::class),
 			]),
+			'translator' => Expect::string()->default(null),
```
Comment (aleswita) [inline src/DI/TranslationExtension.php:56]: Rename to 'translatorFactory'

File: src/DI/TranslationExtension.php [hunk 2]
```diff
@@ -121,6 +122,18 @@ public function loadConfiguration(): void
 			$factory = Contributte\Translation\Translator::class;
 		}
 
+		// Add own Translator
+		if ($this->config->translator !== null) {
+			$reflection = new ReflectionClass($this->config->translator);
+			if (!$reflection->isSubclassOf(Contributte\Translation\Translator::class)) {
+				throw new Contributte\Translation\Exceptions\InvalidArgument('Translator must extends class "' . Contributte\Translation\Translator::class . '".');
+			}
+			if (!$reflection->isSubclassOf(Contributte\Translation\DebuggerTranslator::class)) {
```
Comment (aleswita) [inline src/DI/TranslationExtension.php:131]: Remove this condition, debugger you can set in configuration

File: src/DI/TranslationExtension.php [hunk 3]
```diff
@@ -121,6 +122,18 @@ public function loadConfiguration(): void
 			$factory = Contributte\Translation\Translator::class;
 		}
 
+		// Add own Translator
+		if ($this->config->translator !== null) {
```
Comment (aleswita) [inline src/DI/TranslationExtension.php:126]: Move this condition to row 115 and old condition change to elseif

**Review by aleswita — CHANGES_REQUESTED** [review]

**Comment by aleswita** [comment]
+ write the test

File: src/DI/TranslationExtension.php (hunk 2 above)
Comment (kubomikita) [reply src/DI/TranslationExtension.php:131]: Are you sure ? If I set templateFactory to Translator witch extends Contributte\Translation\Translator and in debug mode (localhost) I get TypeError - Argument 1 passed to Contributte\Translation\Tracy\Panel::__construct() must be an instance of Contributte\Translation\DebuggerTranslator

Comment (aleswita) [reply src/DI/TranslationExtension.php]: You are right, give me some time to make little refactor

File: src/DI/TranslationExtension.php [hunk 4]
```diff
@@ -137,6 +149,9 @@ public function loadConfiguration(): void
 
 		// Tracy\Panel
 		if ($this->config->debug && $this->config->debugger) {
+			if (isset($reflectionTranslatorFactory) && !$reflectionTranslatorFactory->isSubclassOf(Contributte\Translation\DebuggerTranslator::class)) {
```
Comment (aleswita) [inline src/DI/TranslationExtension.php:152]: this you can remove

File: tests/Tests/CustomTranslator.phpt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+$container = require __DIR__ . '/../bootstrap.php';
+
+class CustomTranslator extends Tests\TestAbstract
+{
+
+	public function test01(): void
+	{
+		$translator = new CustomTranslatorMock(new Contributte\Translation\LocaleResolver(Mockery::mock(Nette\DI\Container::class)), new Contributte\Translation\FallbackResolver(), 'en', __DIR__ . '/cacheDir', true);
+
+		Tester\Assert::type(Contributte\Translation\Translator::class, $translator);
```
Comment (aleswita) [inline tests/Tests/CustomTranslator.phpt:20]: move this test to TranslationExtension.phpt to test06

File: tests/Tests/DI/TranslationExtension.phpt [hunk 6]
```diff
@@ -179,6 +179,21 @@ class TranslationExtension extends Tests\TestAbstract
...
+			'translation' => [
+				'locales' => ['whitelist' => ['en']],
+				'translatorFactory' => Tests\CustomTranslatorMock::class,
+			],
+		]);
+
+		/** @var Contributte\Translation\Translator $translator */
+		$translator = $container->getByType(Nette\Localization\ITranslator::class);
+
+		Tester\Assert::type(Contributte\Translation\Translator::class, $translator);
```
Comment (aleswita) [inline tests/Tests/DI/TranslationExtension.phpt:194]: Typo? Here must be **Tests\CustomTranslatorMock**, right?

**Comment by aleswita** [comment]
tnx

**Pull request merged**